src/Cli.Tests/ConfigGeneratorTests.cs

```diff
@@ -119,6 +121,77 @@ public void TestCleanup()
         }
     }
 
+    /// <summary>
+    /// Test to verify creation of initial config with special characters
+    /// such as [!,@,#,$,%,^,&,*, ,(,)] in connection-string or graphql api
+    /// </summary>
+    [TestMethod]
+    public void TestSpecialCharactersInConnectionString()
+    {
+        HandleConfigFileCreationAndDeletion(TEST_RUNTIME_CONFIG_FILE, configFilePresent: false);
+        InitOptions options = new(
+            databaseType: DatabaseType.MSSQL,
+            connectionString: "A!string@with#some$special%characters^to&check*proper(serialization).'",
```

Comment:
If we make this a `DataTestMethod` we can parameterize this connection string, then just re-use that later. Also means if we want to add more cases in the future it is easier.

Reply:
Since this is a cherry pick, looks like it functionally works and the optimization for readability and re-usability can be considered for a future change.